include/openPMD/auxiliary/JSON.hpp

```diff
@@ -23,11 +23,16 @@
 
 #include "openPMD/config.hpp"
 
-#include <nlohmann/json.hpp>
-
 #include <memory>  // std::shared_ptr
+#include <nlohmann/json.hpp>
 #include <utility> // std::forward
 
+#if openPMD_HAVE_MPI
+
+#include <mpi.h>
+
+#endif
```

Comment:
Can you revert the include order for `<nlohmann/json.hpp>`? The one before was right (stdlibs last)

Also please move `mpi.h` before the stdlibs:
```suggestion
#include <nlohmann/json.hpp>

#if openPMD_HAVE_MPI
#   include <mpi.h>
#endif

#include <memory>  // std::shared_ptr
#include <utility> // std::forward
```